### ISSUE TITLE
`/api/github/v1/repository/.../latest` returns a 500 when the provided  branch doesn't exist

### ISSUE THREAD

**Issue by Eijebong** [taskcluster/taskcluster#5851]
Example: `https://{taskcluster_root}/api/github/v1/repository/divvun/kbdi/foobar/latest`

Full error:

```
{"EnvVersion":"2.0","Fields":{"code":422,"incidentId":"2441a83a-0255-46c5-9018-f243daff0bb6","message":"No commit found for SHA: main","method":"latest","name":"HttpError","stack":"HttpError: No commit found for SHA: main\n    at /app/node_modules/@octokit/request/dist-node/index.js:86:21\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Job.doExecute (/app/node_modules/bottleneck/light.js:405:18)","status":422,"v":1},"Hostname":"taskcluster-github-web-566f99f899-vlkzc","Logger":"taskcluster.github.api","Pid":1,"Severity":3,"Timestamp":1670501952432000000,"Type":"monitor.error","message":"HttpError: No commit found for SHA: main\n    at /app/node_modules/@octokit/request/dist-node/index.js:86:21\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Job.doExecute (/app/node_modules/bottleneck/light.js:405:18)","requestId":"5c375a20-c6cf-456a-a8fd-b2c93f266b3a","serviceContext":{"service":"github","version":"45.0.0"},"severity":"ERROR","traceId":"79a7e597-a91c-4cc2-8468-667e734d3798"}
```

It should probably return a 404 with a message telling the user the branch doesn't exist. Note that it already does that when the provided repo doesn't exist.

**Comment by lotas** [comment]
Thanks for reporting this!

Added a fix in #5860, apparently github returns `422` for unknown branches.
Please there was a path when no installation exists for a given repo, so it would just hang and return 500 later because there was no response at all

**Comment by lotas** [comment]
Hmm. I still see 502 on community-tc
```
curl -I https://community-tc.services.mozilla.com/api/github/v1/repository/taskcluster/taskcluster/main2/latest
HTTP/2 502 
date: Wed, 14 Dec 2022 17:12:08 GMT
alt-svc: clear
```
will need to investigate more
